gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/main/java/io/gravitee/repository/elasticsearch/v4/analytics/adapter/EventMetricsQueryAdapter.java

```diff
@@ -217,20 +217,23 @@ private static void applyTopMetricsAggregation(ObjectNode container, String fiel
 
     private static ObjectNode applyFixedIntervalAggregations(HistogramQuery query, ObjectNode aggregationsNode) {
         TimeRange timeRange = query.timeRange();
-        Duration duration = timeRange
+        Duration interval = timeRange
             .interval()
             .orElseThrow(() -> new IllegalArgumentException("Interval is required for TREND aggregations"));
         long from = timeRange.from().toEpochMilli();
         long to = timeRange.to().toEpochMilli();
+        long intervalMillis = interval.toMillis();
+        long alignedMin = (from / intervalMillis) * intervalMillis; // floor
+        long alignedMax = ((to + intervalMillis - 1) / intervalMillis) * intervalMillis; // ceil
 
         ObjectNode intervalNode = aggregationsNode.putObject(PER_INTERVAL);
         ObjectNode histogramNode = intervalNode.putObject(DATE_HISTOGRAM);
         histogramNode.put(FIELD, TIMESTAMP);
-        histogramNode.put(FIXED_INTERVAL, duration.toMillis() + MILLISECONDS);
+        histogramNode.put(FIXED_INTERVAL, intervalMillis + MILLISECONDS);
         histogramNode.put(MIN_DOC_COUNT, 0);
         ObjectNode timeBoundsNode = histogramNode.putObject(EXTENDED_BOUNDS);
-        timeBoundsNode.put(MIN, from);
-        timeBoundsNode.put(MAX, to);
+        timeBoundsNode.put(MIN, alignedMin);
+        timeBoundsNode.put(MAX, alignedMax);
```

Comment:
I like the idea of keeping the wording about `from` and `to`, but it is a personal preference
```suggestion
        timeBoundsNode.put(MIN, alignedFrom);
        timeBoundsNode.put(MAX, alignedTo);
```